### PR TITLE
Kerberos support

### PR DESCRIPTION
**Pull request by viceradman**
Support for Kerberos(gssapi-wtih-mic) authentication based on http://www.openssh.com/txt/rfc4462.txt

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@jenkinsci/code-reviewers

**Comment by mamamic** [comment]
when is this expected to be reviewed?

**Comment by oleg-nenashev** [comment]
Well, there is no active maintainers of this component. So it may take some some time.
The review is in my medium-priority TODO list, but no ETA by now

**Comment by mamamic** [comment]
@oleg-nenashev thanks nevertheless.

File: src/com/trilead/ssh2/packets/PacketUserauthTokenGssapiWithMic.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		this.host = host;
+		
+		context = new GSSContextKrb5();
+		
+		try 
+		{
+			context.create(this.host);
+		} catch (Exception e) 
+		{
+			e.printStackTrace();
```
Comment (mc1arke) [inline src/com/trilead/ssh2/packets/PacketUserauthTokenGssapiWithMic.java]: Use a logger if you need to show the exception contents. Also don't catch Exception unless the underlying class specifically throws it.

File: src/com/trilead/ssh2/packets/PacketUserauthTokenGssapiWithMic.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+			{
+				try 
+				{
+					token=context.init(token, 0, token.length);
+					if (token.length > 1) //take the first generated token and send it to the server
+						break;
+				} catch (Exception e) 
+				{
+					// TODO Auto-generated catch block
+					e.printStackTrace();
```
Comment (mc1arke) [inline src/com/trilead/ssh2/packets/PacketUserauthTokenGssapiWithMic.java]: More suspicious exception handling

File: src/com/trilead/ssh2/packets/PacketUserauthTokenGssapiWithMic.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+		tw.writeString(GSSAPI_WITH_MIC);
+	    
+		byte[] message = tw.getBytes();
+		byte[] mic;
+		try 
+		{
+			mic = context.getMIC(message, 0, message.length);
+		} catch (Exception e) 
+		{
+			e.printStackTrace();
```
Comment (mc1arke) [inline src/com/trilead/ssh2/packets/PacketUserauthTokenGssapiWithMic.java]: More suspicious exception handling

**Comment by Elisedlund-ericsson** [comment]
This pull request is the only one still missing from my fork for me to be able to drop it and use jenkinsci/trilead-ssh2
Could this be merged soon? Need help?

**Comment by olivergondza** [comment]
FTR, we have a dockerzied KDC that can be used to create integration tests once we are confident this is worth adding.

https://github.com/jenkinsci/acceptance-test-harness/tree/master/src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/KerberosContainer

**Comment by kuisathaverat** [comment]
The exceptions are not properly managed, this PR print the stacktrace instead using a logger, also does not explains why capture Exception, thus in case of some error in those points nobody would have a clue of what happens. Finally, it does not have a test. Oliver commented that there is a  dockerzied KDC that can be used on an ATH. So I do not have any problem to merge this PR if you resolve that stuff.

**Comment by kuisathaverat** [comment]
replaced by https://github.com/jenkinsci/trilead-ssh2/pull/38

**Pull request closed without merging**